src/dash-table/components/Table/index.tsx

```diff
@@ -81,6 +83,16 @@ export default class Table extends Component<PropsWithDefaultsAndDerived> {
             virtual.indices
         );
 
+        const virtual_selected_rows = this.virtualSelectedRows(
+            virtual.indices,
+            selected_rows
+        );
+
+        const viewport_selected_rows = this.viewportSelectedRows(
+            viewport.indices,
+            selected_rows
+        );
```

Comment:
The calculation for both virtual and viewport derived props is the same, only the data changes